src/main/java/org/prebid/server/bidder/insticator/InsticatorBidder.java

```diff
@@ -0,0 +1,264 @@
+                .bidfloor(bidFloorPrice.getValue())
+                .build();
+    }
+
+    private Price resolveBidFloor(BidRequest bidRequest, Imp imp) {
+        final Price initialBidFloorPrice = Price.of(imp.getBidfloorcur(), imp.getBidfloor());
+        return BidderUtil.isValidPrice(initialBidFloorPrice)
+                ? convertBidFloor(initialBidFloorPrice, bidRequest)
+                : initialBidFloorPrice;
+    }
+
+    private Price convertBidFloor(Price bidFloorPrice, BidRequest bidRequest) {
+        final BigDecimal convertedPrice = currencyConversionService.convertCurrency(
+                bidFloorPrice.getValue(),
+                bidRequest,
+                bidFloorPrice.getCurrency(),
+                DEFAULT_BIDDER_CURRENCY);
+
+        return Price.of(DEFAULT_BIDDER_CURRENCY, BidderUtil.roundFloor(convertedPrice));
+    }
+
+    private BidRequest modifyRequest(BidRequest request, String publisherId, List<BidderError> errors) {
+        return request.toBuilder()
+                .site(modifySite(request.getSite(), publisherId))
+                .app(modifyApp(request.getApp(), publisherId))
+                .ext(modifyExtRequest(request.getExt(), errors))
+                .build();
+    }
+
+    private static Site modifySite(Site site, String id) {
+        return Optional.ofNullable(site)
+                .map(Site::toBuilder)
+                .map(builder -> builder.publisher(modifyPublisher(site.getPublisher(), id)))
+                .map(Site.SiteBuilder::build)
+                .orElse(null);
+    }
+
+    private static App modifyApp(App app, String id) {
+        return Optional.ofNullable(app)
+                .map(App::toBuilder)
+                .map(builder -> builder.publisher(modifyPublisher(app.getPublisher(), id)))
+                .map(App.AppBuilder::build)
+                .orElse(null);
+    }
+
+    private static Publisher modifyPublisher(Publisher publisher, String id) {
+        return Optional.ofNullable(publisher)
+                .map(Publisher::toBuilder)
+                .orElseGet(Publisher::builder)
+                .id(id)
+                .build();
+    }
+
+    private ExtRequest modifyExtRequest(ExtRequest extRequest, List<BidderError> errors) {
+        final ExtRequest modifiedExtRequest = extRequest == null ? ExtRequest.empty() : extRequest;
+        final InsticatorExtRequest existingInsticator;
+
+        try {
+            existingInsticator = mapper.mapper().convertValue(
+                    modifiedExtRequest.getProperty(INSTICATOR_FIELD),
+                    InsticatorExtRequest.class);
+        } catch (IllegalArgumentException e) {
+            errors.add(BidderError.badInput(e.getMessage()));
+            return modifiedExtRequest;
+        }
+
+        modifiedExtRequest.addProperty(
+                INSTICATOR_FIELD,
+                mapper.mapper().valueToTree(buildInsticator(existingInsticator)));
+
+        return modifiedExtRequest;
+    }
```

Comment:
Based on the code in GO, we should always add `Prebid caller`.
Please, fix this, and check that the logic of other parts of the adapter corresponds the GO version.